### ISSUE TITLE
Make Basket accept names

### ISSUE THREAD

**Issue by stephaniemcv** [MozillaFoundation/encryption_campaign#84]
Is it possible to fix this before Encryption Campaign launch?

From a long email thread - 

Jon Petto: One major point brought to my attention yesterday is that basket doesn’t accept a name (or first name/last name). This means those fields on our form (privacy day) are superfluous, and is likely one reason why /data-retention/ sent information to sendto as well as basket. Thoughts on what we should do here? Easiest thing is removing the first and last name form fields. 

Jess Davis: You can use the send.to endpoint as I believe Adam has a script that then runs on BSD (where the send.to endpoint puts subscribers), and puts new subscribers into ExactTarget (using basket so their assigned a token if they aren't already present; if present, their record in Master_Subscribers will get updated). Ccing Adam here to confirm - and to alert him of this technical gap to be overcome before we can switch over all BSD/send.to forms to directly to basket + ExactTarget. (Me, Adam and Pmac will need to work together to figure out how to solve this. I'll file a bug.)

What using send.to will _not_ allow you to do is trigger an email right away. Any automated email follow up would have to be part of an automation within ET to pull and find these folks, then send them an email.

@adamlofting : BSD talks to basket to keep the subscriptions / unsubcribes in sync, so it has the same name limitation. Our historic records (where we have name data) were loaded directly into Exact Target by DEG in a one time load. We haven't run any petitions since then so don't have new names.

We don't want to be putting new names into BSD, as we need to close it down soon.

Jess Davis: Totally agree on the general importance (of collecting names), and I apologize to the Foundation team that I missed the detail of making sure that Basket could handle incoming First & Last names from forms when we began joining systems last year. (We had a place to store it from manual imports - but not for incoming signups :/)
**
Basket cannot store this info at present - and to make this change it'd be another code push with the steps Pmac has outlined. It's out of scope for this project/timeline.**

We'll address how to capture first & last names from forms with Basket as we work on the email platform re-architecture project this quarter. I'll keep you posted.

**Comment by adamlofting** [comment]
Drive by notes from me:
- Let's invest any time/money/energy in Basket and ET, not BSD
- We can't ship this for campaign launch, but the campaign runs for many weeks with several CTAs. Starting now is valuable, even if we only get the benefit in the 2nd half of the campaign
- The timeline was mostly nudging/chasing/project management and not many hours of dev - Can we (Mofo) help with that?:

Adding the work estimate to this thread for reference:
1. Code the thing including tests. (pretty simple; an hour or two)
2. Get someone to review the changes. (potentially the next day)
3. Merge the commits to master. (same day as above if it's good)
4. File a WebOps bug to have the changes pushed to dev and stage instances. (could be a couple days)
5. Test those changes. (an hour or so)
6. Request that the changes be pushed to prod. (could take another bit before it's picked up again by Ops)
7. Test them again once the prod push has completed. (pretty quick)

**Comment by valianttry** [comment]
The ability to address people by first name in messaging usually has an impact on conversion. First, I'm going to ask MoCo if they are open to let us help make this functionality possible in basket, per @adamlofting 's work estimate.
cc @simonwex @OpenMatt

**Comment by simonwex** [comment]
Looks like Paul made some fast moves here. Hopefully this will get merged tomorrow.

**Comment by OpenMatt** [comment]
thanks @pmclanahan !

@simonwex I'm going to assign to you just to follow up here and make sure this doesn't fall through the cracks, if that's ok.

**Comment by pmclanahan** [comment]
It is merged, and I just pushed the backported version to our `legacy` branch (we have 2 different deployments at the moment, _sigh_). So the changes are already on our [new infra staging instance](https://basket-stage.us-west.moz.works) ([auto deploy](https://travis-ci.org/mozilla/basket/builds/105682713)) and we can test there. I've also [filed the bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1244115) to push the changes to our legacy stage and (after testing) prod instances. I'll try to prod webops a bit to see if we can get this expedited. In the mean time we can test against the new infra if needed

**Comment by jessilyndavis** [comment]
Nice work @pmclanahan! Thank you!

@adamlofting , @stephaniemcv  - please make sure that the new functionality is fully tested to ensure:

1) The first & last names go through to ExactTarget using _new_ email addresses that have never been in ET before. 
- You'll need to look for these individuals in the "Double_Opt_In" data extension. (They won't be added to "Master_Subscribers" until the subscriber confirms the signup via the confirmation email, and the nightly process runs within ET at midnight.)

2) That signup forms still work.

Ben and I are unable to help with testing. Please make sure that this is fully tested before it goes live. If you have questions, I can help answer them here or in IRC.

Thanks!

**Comment by pmclanahan** [comment]
Well! This is great if a little embarrassing given my dire time predictions above: IT'S DONE! Ops was able to hop on it very quickly, I got it tested in stage, and it's now in Prod and confirmed passing through first and last names :fireworks: So, along with your normal basket subscription calls you can now send `first_name` and `last_name` as fields and they'll be stored in ExactTarget.

Cheers! :beers:

**Comment by jessilyndavis** [comment]
Well hot damn! Nice work @pmclanahan !  I just tested prod real quick to make sure that the Firefox + You newsletter form works - and it does!

Foundation team - it's up to you to do the testing of your forms to make sure the right info is being passed through to ET.

**Comment by stephaniemcv** [comment]
@adamlofting Do we have a list of all our signup form urls?

**Comment by OpenMatt** [comment]
@pmclanahan and @jessilyndavis thank you so much! Amazing news. 
@stephaniemcv I assume you are taking point on testing for MoFo, so assigning to you.

**Comment by OpenMatt** [comment]
@jbuck can you help test this for us?

**Comment by jbuck** [comment]
@stephaniemcv I signed up to production with the email `jon+testing1001@mozillafoundation.org`, the first_name `Jon Testing 1001`, and the last_name `Buckley` to the Foundation mailing address. Could you confirm that I have my first name and last name somewhere in ET?

**Comment by stephaniemcv** [comment]
I confirm that I see jon+testing1001@mozillafoundation.org

**Comment by jbuck** [comment]
@stephaniemcv but do you see the first name and last name?

**Comment by jbuck** [comment]
@pmclanahan was able to confirm that my first name and last name was entered correctly, so we're good here!

**Comment by OpenMatt** [comment]
thanks @jbuck !
